virtual/activations.go

```diff
@@ -382,6 +383,46 @@ func (a *activations) getServerState() (
 	return a.serverState.serverID, a.serverState.serverVersion
 }
 
+func (a *activations) close(ctx context.Context, numWorkers int) error {
+	a.Lock()
+	defer a.Unlock()
+
+	// create a channel to receive keys
+	keys := make(chan types.NamespacedActorID, len(a._actors))
+	for k := range a._actors {
+		keys <- k
+	}
+	close(keys)
+
+	// create a wait group to wait for all goroutines to finish
```

Comment:
Maybe... hahaha just two of them were. The others I added them myself. I used chatgpt to ask what's the most efficient way to spawn multiple goroutines, in case she could surprise me.